recipes/sipbuildtool/conanfile.py

```diff
@@ -0,0 +1,140 @@
+import os
+import sys
+
+from pathlib import Path
+from io import StringIO
+
+from jinja2 import Template
+
+from conan import ConanFile
+from conan.tools.env import VirtualRunEnv
+from conan.tools.files import files
+from conans.errors import ConanException
+
+
+class SipBuildTool(object):
+    """
+    A generator that creates a CMake script to build python bindings for
+    projects using Sip.
+
+    This generator expects the project to have a pyproject.toml.jinja file that
+    determines which Sip files to compile. It configures that file to a TOML
+    file that Sip can use.
+
+    Usage in a conanfile for a Sip project:
+    sip = self.python_requires["sipbuildtool"].module.SipBuildTool(self)
+    sip.configure()
+    sip.generate("projectName")
+    """
+    def __init__(self, conanfile: ConanFile):
+        self.conanfile = conanfile
+        env = VirtualRunEnv(self.conanfile)
+        self.run_env = env.environment()
+        self.envvars = None
+        self.python_venv_interpreter = None
+        self.sip_build_executable = None
+
+    @property
+    def _venv_base_path(self):
+        return os.path.join(self.conanfile.build_folder, "venv")
+
+    @property
+    def _venv_bin_path(self):
+        if self.conanfile.settings.os == "Windows":
+            return os.path.join(self._venv_base_path, "Scripts")
+        return os.path.join(self._venv_base_path, "bin")
+
+    def configure(self, sip_version = "6.5.1", python_interpreter = sys.executable):
+        python_interpreter = Path(python_interpreter)
+
+        # When on Windows execute as Windows Path
+        if self.conanfile.settings.os == "Windows":
+            python_interpreter = Path(*[f'"{p}"' if " " in p else p for p in python_interpreter.parts])
```

Comment:
Paths without spaces will work fine with or without quotes around. I the code becomes more readable if the quotes are removed
```suggestion
            python_interpreter = Path(*[f'"{p}"' for p in python_interpreter.parts])
```

Reply:
**Touch this at your own peril**
Somehow windows PowerShell and python do weird things together.

It took me ages to handle those paths correctly and shell and OS agnostic and it is a hell of a process to get this correct over 3 OSes and two different shell languages 

https://cpplang.slack.com/archives/C41CWV9HA/p1654942697502539?thread_ts=1654942697.502539&cid=C41CWV9HA